### PR TITLE
Feature/personas

### PR DESCRIPTION
**Pull request by AnniePachecoJ**
# Summary
This PR modifies the object Persona and PersonaCreate, changing the fields voiceId to Voice object and faceId to Face object

 
# Changes
 
- [x] Added Voice schema
- [x] Added Face schema
- [x] Modified Persona
- [x] Modified PersonaCreate

 
# Related Issues
N/A
 
# Type of Change
This change solved the comments posted on the review of the previous PR created by @rquesada [https://github.com/pAI-OS/paios/pull/4]
 
# Checklist 
- [x] I have read the [CONTRIBUTING](https://github.com/pAI-OS/paios/blob/main/CONTRIBUTING.md) documentation.
- [x]  I have performed a self-review of my own code.
- [x]  I have added tests that prove my fix is effective or that my feature works.
- [x]  I have updated the documentation (if applicable).
 
# Screenshots (if applicable)
image
image
image
 
 
# Additional Context
Tested locally using Swagger.
Based on the premise that we are not expecting that faces and voices to be modified frequently, so  the approach we considered here was to create an "assets" folder (in the front-end) to store the images, videos and audios. Then it was only going to be necessary to store the id of voice and face to help us associate the asset with the voice and face.

For example:
Given the object Persona:
      {
        "name": "Mari",
        "description": "Mari A",
        "voiceId": "x4te5a69-40d6-4246-8d53-f4c777be8080",
        "faceId": "7ffe5a69-40d6-4246-8d54-f4c777be6989",
        "id": "91a7542f-da52-46ab-85b9-08f871722e93"
      }

The frontend will be able to load the files:
      :/assests/audio/x4te5a69-40d6-4246-8d53-f4c777be8080.mp3
      :/assests/face/7ffe5a69-40d6-4246-8d54-f4c777be6989.jpg
      :/assests/video/7ffe5a69-40d6-4246-8d54-f4c777be6989.mp4

### CODE REVIEW

**Review by quaid — CHANGES_REQUESTED** [review]
Hey @AnniePachecoJ thanks for the review request, happy to do what I can.

Since I'm not (yet) familiar with the project coding standards and I'm not a programmer, there's not much I have to say about the code changes. It doesn't look like you added any tests, documentation, or many additional comments in the code itself? For many projects I'd expect to see signs of test driven development and docs or code comments similar to the "Additional Context" in the PR template. But I'm not blocking on those, and I don't know if those are blockers.

For the change I'm requesting, I may be incorrect so please let me know, but GitHub is marking the PR as blocked because the commit was not signed. IIUC, you can create and self-sign a GPG key, then configure git locally and GitHub to use sign your commit, and that should be sufficient. All of the steps with keysigning and so forth shouldn't be part of the signing of the commit, rather those are part of building the web of trust within the project.

Here's the link GH gives from the blocked merge area:

https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

File: apis/paios/openapi.yaml [hunk 1]
```diff
@@ -1074,8 +1074,32 @@ components:
...
           $ref: '#/components/schemas/textLong'
         voiceId:
-          $ref: '#/components/schemas/uuid4'
+          $ref: '#/components/schemas/textShort'
         faceId:
-          $ref: '#/components/schemas/uuid4'
+          $ref: '#/components/schemas/textShort'
+      required:        
+        - name
+    Voice:
```
Comment (rquesada) [inline apis/paios/openapi.yaml:1082]: Remove Voice schema, it is unused

File: apis/paios/openapi.yaml [hunk 2]
```diff
@@ -1074,8 +1074,32 @@ components:
...
+        id:
+          $ref: '#/components/schemas/textShort'
+        name:
+          $ref: '#/components/schemas/textShort'
       required:
-        - name
\ No newline at end of file
+        - id
+        - name
+    Face:
```
Comment (rquesada) [inline apis/paios/openapi.yaml:1093]: Remove Face schema, it is unused

File: backend/api/PersonasView.py [hunk 3]
```diff
@@ -15,17 +13,17 @@ async def get(self, id: str):
...
         persona = await self.pm.retrieve_persona(id)
-        return JSONResponse(persona.model_dump(), status_code=201, headers={'Location': f'{api_base_url}/personas/{id}'})
+        return JSONResponse(persona.dict(), status_code=201, headers={'Location': f'{api_base_url}/personas/{id}'})
 
     async def put(self, id: str, body: PersonaCreateSchema):
-        await self.pm.update_persona(id, **body.dict())
+        await self.pm.update_persona(id, body)
         persona = await self.pm.retrieve_persona(id)
-        return JSONResponse(persona.model_dump(), status_code=200)
+        return JSONResponse(persona.dict(), status_code=200)
```
Comment (rquesada) [inline backend/api/PersonasView.py:26]: add validation if person is None like:

```
if persona is None:
     return JSONResponse({"error": "Persona not found"}, status_code=404)
return JSONResponse(persona.dict(), status_code=200)
```

File: backend/api/PersonasView.py [hunk 4]
```diff
@@ -15,17 +13,17 @@ async def get(self, id: str):
...
 
     async def delete(self, id: str):
         await self.pm.delete_persona(id)
```
Comment (rquesada) [inline backend/api/PersonasView.py:29]: Add Validation:
```
success = await self.am.delete_persona(id)
if not success:
      return JSONResponse({"error": "Persona not found"}, status_code=404)
```

File: backend/managers/PersonasManager.py [hunk 5]
```diff
@@ -22,16 +22,17 @@ def __init__(self):
...
-            await session.commit()
-            return new_persona.id
+            await session.commit() 
+            await session.refresh(new_persona)
+            return new_persona.id    
 
     async def update_persona(self, id: str, persona_data: PersonaCreateSchema):
         async with db_session_context() as session:
-            stmt = update(Persona).where(Persona.id == id).values(**persona_data.model_dump())
+            stmt = update(Persona).where(Persona.id == id).values(**persona_data)
```
Comment (rquesada) [inline backend/managers/PersonasManager.py:35]: add validation:
```
result = await session.execute(stmt)
if result.rowcount > 0:                
      await session.commit()
      updated_persona = await session.get(Persona, id)
      return PersonaSchema(id=updated_persona.id, **persona_data)
return None
```

File: backend/managers/PersonasManager.py [hunk 6]
```diff
@@ -41,11 +42,19 @@ async def delete_persona(self, id):
             await session.execute(stmt)
             await session.commit()
```
Comment (rquesada) [inline backend/managers/PersonasManager.py:43]: Add:
```
await session.commit()
return result.rowcount > 0
```

File: backend/managers/PersonasManager.py [hunk 7]
```diff
@@ -41,11 +42,19 @@ async def delete_persona(self, id):
...
+                return PersonaSchema(
+                    id=persona.id,
+                    name=persona.name,
+                    description=persona.description,
+                    voiceId=persona.voiceId,
+                    faceId=persona.faceId                    
+                )
+            return None
 
     async def retrieve_personas(self, offset=0, limit=100, sort_by=None, sort_order='asc', filters=None):
```
Comment (rquesada) [inline backend/managers/PersonasManager.py:59]: Add a return value:
```
async def retrieve_personas(self, offset:int = 0, limit:int = 100, sort_by:Optional[str] = None, sort_order:str = 'asc',  [filters=None](filters: Optional[Dict[str, Any]] = None)): -> Tuple[List[PersonaSchema], int]
```

File: backend/managers/PersonasManager.py [hunk 8]
```diff
@@ -22,16 +22,17 @@ def __init__(self):
...
-            new_persona = Persona(id=str(uuid4()), **persona_data.model_dump())
+            new_persona = Persona(id=str(uuid4()), **persona_data)
             session.add(new_persona)
-            await session.commit()
-            return new_persona.id
+            await session.commit() 
+            await session.refresh(new_persona)
+            return new_persona.id    
 
     async def update_persona(self, id: str, persona_data: PersonaCreateSchema):
```
Comment (rquesada) [inline backend/managers/PersonasManager.py:33]: add a return value:
```
async def update_persona(self, id: str, persona_data: PersonaCreateSchema): -> Optional[PersonaSchema]
```

File: backend/managers/PersonasManager.py [hunk 9]
```diff
@@ -22,16 +22,17 @@ def __init__(self):
...
         async with db_session_context() as session:
-            stmt = update(Persona).where(Persona.id == id).values(**persona_data.model_dump())
+            stmt = update(Persona).where(Persona.id == id).values(**persona_data)
             await session.execute(stmt)
             await session.commit()
```
Comment (rquesada) [inline backend/managers/PersonasManager.py:38]: add return value:
```
async def delete_persona(self, id): -> bool
```

File: backend/managers/PersonasManager.py [hunk 10]
```diff
@@ -41,11 +42,19 @@ async def delete_persona(self, id):
...
 
-    async def retrieve_persona(self, id):
-        async with db_session_context() as session:
+    async def retrieve_persona(self, id:str) -> PersonaSchema:
```
Comment (rquesada) [inline backend/managers/PersonasManager.py:45]: Fix return value:
```
async def retrieve_persona(self, id:str) -> Optional[PersonaSchema]:
```

File: backend/models.py [hunk 11]
```diff
@@ -35,3 +35,13 @@ class Persona(Base):
     voiceId = Column(String, nullable=True)
     faceId = Column(String, nullable=True)
 
+class Voice(Base):
```
Comment (rquesada) [inline backend/models.py:38]: Remove unused Voice model

File: backend/models.py [hunk 12]
```diff
@@ -35,3 +35,13 @@ class Persona(Base):
...
+    __tablename__ = "voice"
+    id = Column(String, primary_key=True)
+    voice_id = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+
+class Face(Base):
```
Comment (rquesada) [inline backend/models.py:44]: Remove unused Face model

File: backend/schemas.py [hunk 13]
```diff
@@ -59,3 +63,25 @@ class AssetCreateSchema(AssetBaseSchema):
 
 class AssetSchema(AssetBaseSchema):
     id: str
+
```
Comment (rquesada) [inline backend/schemas.py:66]: remove unused class:
-VoiceBaseSchema
-VoiceCreateSchema
-VoiceSchema
-FaceBaseSchema
-FaceCreateSchema
-FaceSchema

**Review by rquesada — CHANGES_REQUESTED** [review]
Add some feedback

**Comment by AnniePachecoJ** [comment]
This PR has to be deleted because there were some commits which were unverified

**Pull request closed without merging**